pkgs/development/python-modules/pur/default.nix

```diff
@@ -1,19 +1,26 @@
 { lib
-, buildPythonPackage
-, click
+, python3
 , fetchFromGitHub
-, pytestCheckHook
 }:
 
+let
+  py = python3.override {
+    packageOverrides = self: super: {
+      click = self.callPackage ../../../development/python-modules/click/7.nix { };
+    };
+  };
+in
+with py.pkgs;
```

Comment:
The other usages of this approach on `master` are fine because they're apps (not Python modules). However we probably can't use the same approach here since this is causing `python38Packages.pur` to be build with Python 3.9.

Reply:
Yeah, I think you're right. If I get rid of the override part here and just do click = callpackage ... directly, the python38 package seems to be correct.

Reply:
Maybe we should have a `python3Packages.click_7` and make this package use `click_7` instead of `click`, so if other packages needs this in the future they can just use the same input.

However I am not sure how we decide if we will expose something on `pkgs/top-level/python-packages.nix`. Maybe @SuperSandro2000 can help us here.

Reply:
We just add a new line that points to 7.nix instead of default? This patch (apply ontop of this PR) http://0x0.st/-5jO.txt does that, builds fine and results in different hashes for the two python versions.

Reply:
This is not allowed in pythonPackages because python cannot mix multiple versions of the same package and things break in really strange ways when it happens.